scope.go

```diff
@@ -579,6 +579,23 @@ type clusterUser struct {
 	isWildcarded    bool
 }
 
+func deepCopy(cu *clusterUser) *clusterUser {
```

Comment:
Can you explain why is it needed?

Reply:
because the current implementation doesn't do a deepcopy and only modify the name & pwd of the cu user.
So, if analyst_john-UK & analyst_jane-UK are doing a query  at the same time we can end-up in this situation:
- cu user name = john & password = pwd1
- cu user name = jane & password = pwd2
- the query of john is executed in clickhouse with the credentials jane/pwd2
- the query of jane is executed in clickhouse with the credentials jane/pwd2